### PR TITLE
Mailchimp: signup button

### PR DESCRIPTION
**Pull request by artpi**
zrzut ekranu 2019-02-28 o 18 25 32

We are waiting for the link to put in the form

### CODE REVIEW

**Comment by cburton4** [comment]
Hey @artpi,

It's exciting to see this Mailchimp integration happening! Great work 😎 

I'm swinging by to leave a bit of design feedback for buttons as the Muriel guidance might not be the easiest to find or digest just yet. My first note is in regards to visual hierarchy for buttons. The Muriel guidance for this states:

> A layout should contain a single prominent button that makes it clear that other buttons have less importance in the hierarchy. Use a high-emphasis contained button to command the most attention.

In layman's terms this means that there should only be one contained button (also known as button primary / pink button) per screen and whichever button is contained should be the most important action we want a user to take on the screen. If all buttons have the same level of importance they should either be outlined buttons or text buttons. 

hierarchy-pyramid-1

More guidance on [button hierarchy and placement](https://murieldesignsystem.blog/components/buttons/#hierarchy-and-placement).

The second note is in regards to the toggle button (also known here as split button). Toggle buttons should only be used when you're wanting a customer to select an option that de-selects another.

> Only one option in a group of toggle buttons can be selected and active at a time. Selecting one option deselects any other.

mio-design2fassets2f1p-qidsjdzpaq2osum72hnfwfayfldikp2fbuttons-toggle-usage

More guidance for using the [toggle button](https://murieldesignsystem.blog/components/buttons/toggle-button/).

My suggestion would be to make all of the contained buttons (primary buttons) outlined buttons and for the toggle button to be separated into two separate buttons.

screen shot 2019-03-04 at 11 33 31 pm

Does this make sense? Do you have any questions about any of this?

**Comment by davemart-in** [comment]
Thanks @cburton4! I'll post an update once I've had some more time to look at this.

**Comment by davemart-in** [comment]
@artpi I made some updates in https://github.com/Automattic/wp-calypso/commit/f304fab55466170903528dcfe501811d9330fdf6 based on @cburton4's Muriel feedback. Here's a before/after:

## Before



## After

File: client/my-sites/sharing/style.scss [hunk 1]
```diff
@@ -548,7 +548,7 @@
 	margin-bottom: 20px;
 	padding: 20px 24px;
 	background: var( --color-white );
-	box-shadow: 0 0 0 1px rgba( var( --color-neutral-100-rgb ), 0.5 ), 0 1px 2px var( --color-neutral-0 );
+	box-shadow: 0 0 0 1px rgba( lighten( $gray, 20% ), 0.5 ), 0 1px 2px var( --color-neutral-0 );
```
Comment (simison) [inline client/my-sites/sharing/style.scss]: I don't think we should be using SASS colour variables now that we have themes. cc @flootr

**Comment by simison** [comment]
Noting that primary/secondary button update for this page could be PRed separately from this PR so that it can land before we are ready with Mailchimp "sign up" link?

File: client/my-sites/sharing/style.scss (hunk 1 above)
Comment (davemart-in) [reply client/my-sites/sharing/style.scss]: You're right. Good eye. Thanks for calling this out. Reverted in https://github.com/Automattic/wp-calypso/commit/defda7181fc6d34534d46798b5211444e8c180f2

**Comment by davemart-in** [comment]
> Noting that primary/secondary button update for this page could be PRed separately from this PR so that it can land before we are ready with Mailchimp "sign up" link?

Good call. I've reverted this change in https://github.com/Automattic/wp-calypso/commit/defda7181fc6d34534d46798b5211444e8c180f2 and will submit this in a separate PR.

**Comment by davemart-in** [comment]
Here's an updated screenshot of the current state of things:

**Comment by davemart-in** [comment]
Update: Here's the separate PR for removing the primary button states: https://github.com/Automattic/wp-calypso/pull/31263

File: client/my-sites/sharing/connections/service-action.jsx [hunk 2]
```diff
@@ -83,10 +83,22 @@ const SharingServiceAction = ( {
 	if ( primary && 'mailchimp' === service.ID ) {
 		return (
 			<div>
-				<Button className="connections__signup" compact>
+				<Button
+					className="connections__signup"
+					compact
+					href="https://public-api.wordpress.com/rest/v1.1/sharing/mailchimp/signup"
+					target="_blank"
```
Comment (simison) [inline client/my-sites/sharing/connections/service-action.jsx]: Just making sure this button is returning `rel="noopener noreferrer"` since those are needed for external _blank targeted links?

Comment (artpi) [reply client/my-sites/sharing/connections/service-action.jsx]: Button is smart like that https://github.com/Automattic/wp-calypso/blob/master/client/components/button/index.jsx#L40

Comment (simison) [reply client/my-sites/sharing/connections/service-action.jsx]: Excellent, thanks!

**Comment by simison** [comment]
Make sure to rebase with `master` just in case, as the PR has been open quite a while.

**Comment by simison** [comment]
Suppose color changes from https://github.com/Automattic/wp-calypso/pull/31263 start working once rebased.

**Comment by artpi** [comment]
Just rebased, waiting for tests
